public/app/features/dashboard-scene/panel-edit/PanelEditor.tsx

```diff
@@ -147,3 +145,52 @@ export function buildPanelEditScene(panel: VizPanel): PanelEditor {
     }),
   });
 }
+
+// This function is used to conditionally add the data pane to the panel editor,
+// depending on the type of a panel being edited.
+function conditionalDataPaneBehavior(scene: SplitLayout) {
+  const dashboard = getDashboardSceneFor(scene);
+
+  const editor = dashboard.state.editPanel;
+
+  if (!editor) {
+    return;
+  }
+
+  const panelManager = editor.state.panelRef.resolve();
+  const panel = panelManager.state.panel;
+
+  if (!config.panels[panel.state.pluginId].skipDataQuery) {
+    scene.setState({
+      secondary: new SceneFlexItem({
+        body: new PanelDataPane(panelManager),
+      }),
+    });
+  }
+
+  const sub = panelManager.subscribeToState((n, p) => {
+    const nextPluginId = n.panel.state.pluginId;
+    const prevPluginId = p.panel.state.pluginId;
+    const hadDataSupport = !config.panels[prevPluginId].skipDataQuery;
+    const willHaveDataSupport = !config.panels[nextPluginId].skipDataQuery;
+
+    if (nextPluginId !== prevPluginId) {
+      if (hadDataSupport && !willHaveDataSupport) {
+        locationService.partial({ tab: null }, true);
+        scene.setState({
+          secondary: undefined,
+        });
+      } else if (!hadDataSupport && willHaveDataSupport) {
+        scene.setState({
+          secondary: new SceneFlexItem({
+            body: new PanelDataPane(panelManager),
+          }),
+        });
+      }
+    }
+  });
```

Comment:
Think this should be the same and a bit simpler/smaller: 

```suggestion
  const sub = panelManager.subscribeToState((n, p) => {
    const hadDataSupport = !config.panels[n.panel.state.pluginId].skipDataQuery;
    const willHaveDataSupport = !config.panels[p.panel.state.pluginId].skipDataQuery;
     
      if (hadDataSupport && !willHaveDataSupport) {
        locationService.partial({ tab: null }, true);
        scene.setState({  secondary: undefined    });
      } else if (!hadDataSupport && willHaveDataSupport) {
        scene.setState({ secondary: dataPane });      
     }
  });
```